[HCCDOC#1765]: Add recommendations-filtering content to Remote Health Monitoring

kblake2016 opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.11+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: https://issues.redhat.com/browse/HCCDOC-1765
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview:
Two sections added
- https://70364--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster#insights-operator-advisor-recommendation-filters_using-insights-to-identify-issues-with-your-cluster
- https://70364--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster#insights-operator-using-filters-on-advisor-recommendations_using-insights-to-identify-issues-with-your-cluster

<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [X ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

kblake2016 commented:
@jholecek-rh; because you provided the most feedback for this best-practice filtering content, please ack the QE review.

kblake2016 commented:
/label peer-review-needed

shipsing commented:
/label peer-review-in-progress

shipsing commented:
/remove-label peer-review-needed

shipsing commented on modules/insights-operator-using-filters-on-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,32 @@
+
+.Procedure
+. Navigate to the https://console.redhat.com/openshift/insights/advisor/recommendations[Red Hat Hybrid Cloud Console > OpenShift > Advisor recommendations] page.
+. Using the filter input fields above the list of recommendations, select the *Category* primary filter.
+. Expand the secondary filter dropdown and check the box next to each category of recommendation you want to view. Leave unnecessary categories unchecked.
+. Optionally, add additional filters to further refine the list.
+
```
We can make this procedure generic and simple, rather sepcifying any specific filter like Category.

. From the *filter types* list that is present above the list of recommendations, select a filter.
. Set the value of the filter by selecting the relevant options from the drop-down list.
. Optional: Apply additional filters by performing the same steps to further refine the recommendations list.

shipsing commented on modules/insights-operator-using-filters-on-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,32 @@
+. Expand the secondary filter dropdown and check the box next to each category of recommendation you want to view. Leave unnecessary categories unchecked.
+. Optionally, add additional filters to further refine the list.
+
+Only recommendations from the selected categories are shown in the list. 
```
We can remove this line to make this procedure generic.

shipsing commented on modules/insights-operator-using-filters-on-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,32 @@
+
+As an OpenShift cluster manager, you can filter the recommendations that are displayed on the recommendations list. You might do this to reduce the number of reported recommendations or to concentrate on your highest priority issues. 
+
+Use the following example to select from several categories of recommendations to hide unneeded results.
```
We can remove this sentence and try to make the below procedure generic, rather making it specific by applying only the Category filter.

shipsing commented on modules/insights-operator-advisor-recommendation-filters.adoc:
```diff
@@ -0,0 +1,29 @@
+
+You can set the following filters:
+
+* *Name:* Search for a recommendation by name.
+* *Total risk:* An abstract value derived from the likelihood that the detected issue will negatively affect your infrastructure, and the impact on operations if that were to happen.
+Select one or more values from Critical, Important, Moderate, and Low.
+* *Impact:* The potential impact to the continuity of cluster operations. Select from Critical, High, Medium, and Low.
+* *Likelihood:* Filter by the potential for negative impact to clusters should the consequences of the recommendation come to fruition. Critical, High, Medium, or Low.
+* *Category:* Select one or more categories of recommendations to concentrate on: Service Availability, Performance, Fault Tolerance, Security, Best Practice.
+* *Status:* Set to only show Enabled recommendations by default, you can set the filter to show disabled recommendations.
+* *Clusters impacted:* Also set by default, you can filter to only recommendations currently impacting one or more clusters, and or show non-impacting recommendations.
+* *Risk of change:* A value that indicates the risks associated with carrying out the resolution steps and their impact on cluster operations.
```
Can we simplify the description of the the `Likelihood` filter type in the above list?

shipsing commented on modules/insights-operator-using-filters-on-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,32 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="insights-operator-using-filters-on-advisor-recommendations_{context}"]
+= Using filters to remove unnecessary results from Insights advisor recommendations
```
Rename this procedure module to `Applying a filter` to make it simple and clear.

shipsing commented on modules/insights-operator-using-filters-on-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,32 @@
+Only recommendations from the selected categories are shown in the list. 
+
+.Verification
+You can see the applied filters, including default filters, below the filter input dropdowns and above the list of recommendations.
```
Simplify the 

```suggestion
* After applying filters, you can view the updated recommendations list. The applied filters are added next to the default filters.
```

shipsing commented on modules/insights-operator-using-filters-on-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,32 @@
+
+.Removing filters
+
+* You can remove applied filters individually, including the default filters, by clicking the *X* next to the filter, where applied filters are displayed.
+* You can remove only the filters that you applied, leaving the default filters intact, by clicking the *Remove filters* button.
```
```suggestion
You can apply multiple filters to the list of recommendations and remove them individually by performing one of the following actions:
* Click the clear icon next to the filter.
* Click *Reset filters* to return to the default view without any manually applied filters.
```
I do not see the *Remove filters* button in the GUI. Please check the second bullet point for correctness.

shipsing reviewed (changes requested):
@, I have provided a few suggestions for you to consider. Please have a look at them.
 
Since i am new to the peer review squad, i would ask @bburt-rh to give my review the go-ahead. Please wait until Brain applies the peer-review-done label.

kblake2016 commented:
@shipsing , thank you for the helpful feedback. I implemented it. I changed file names and titles/headings to simplify things. However, now I'm seeing a asciidoc validation error, which I think is due to changing one of the file names. Can you advise how to proceed?

kblake2016 commented:
/test validate-asciidoc

xJustin commented on support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.adoc:
```diff
@@ -15,6 +15,9 @@ include::modules/insights-operator-advisor-overview.adoc[leveloffset=+1]
 include::modules/displaying-potential-issues-with-your-cluster.adoc[leveloffset=+1]
 include::modules/displaying-all-insights-advisor-recommendations.adoc[leveloffset=+1]
+include::modules/insights-advisor-recommendations-filters.adoc[leveloffset=+1]
+include::modules/filtering-insights-advisor-recommendations.adoc.[leveloffset=+2]
```
```suggestion
include::modules/filtering-insights-advisor-recommendations.adoc[leveloffset=+2]
```

xJustin reviewed (commented):
Hey @kblake2016 I think I found your problem.

bburt-rh commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+.Procedure
+. Navigate to the link:https://console.redhat.com/openshift/insights/advisor/recommendations[*Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*] page.
+. Using the filter input fields above the list of recommendations, select the *Category* filter type.
+. Expand the right dropdown and check the box next to each category of recommendation you want to view. Leave unnecessary categories unchecked.
```
```suggestion
. From the [add label name!] drop-down list, select each category of recommendation you want to view.
```

- [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#location-of-interface-elements), "When you refer to the location of an element in a graphical user interface, do not use sensory words, such as _top_, _bottom_, _right_, or _left_." 
- Use [the ISG standards for drop-down lists](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#drop_down_list) whenever you refer to these UI elements.
- Per the ISG, do not use "check" to refer to marking a checkbox. Use "select".
- Also if it is a checkbox, refer to it as a "checkbox" and not a "box".

kblake2016 replied:
Done. Just as an fyi, the dropdowns on the Hybrid Cloud Console filters don't have label names (or hover-over names); there are two of them, and they're side by side. We call them the primary and secondary filters in another Console project, but neither the engineers nor peer review like that phrasing so I'm trying to be specific for the users, given both the limitations and rapidly changing nature of the GUI.

bburt-rh commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+. Navigate to the link:https://console.redhat.com/openshift/insights/advisor/recommendations[*Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*] page.
+. Using the filter input fields above the list of recommendations, select the *Category* filter type.
+. Expand the right dropdown and check the box next to each category of recommendation you want to view. Leave unnecessary categories unchecked.
+. Optionally, add additional filters to further refine the list.
```
```suggestion
. Optional: Add additional filters to further refine the list.
```
Follow [the ISG guidance](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps) for indicating optional steps.

kblake2016 replied:
Done

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+To apply a filter, select a filter type and then set its value based on the options that are available in the drop-down list. You can apply multiple filters to the list of recommendations.
+
+You can set the following filter types:
+
```
Not sure about this, but maybe the items here would be easier to read at a glance if they were in a table rather than a bulleted list? WDYT?

In any case, per [the ISG guidance on the wording of list items](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#wording-of-list-items), all of the first sentences of the descriptions must be made grammatically parallel. Re-write so that either they all begin with an imperative "Search for...", "Filter by.." to form complete sentences or they begin with a descriptive sentence fragment (as in the "Total risk" item).

kblake2016 replied:
done

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+You can set the following filter types:
+
+* *Name:* Search for a recommendation by name.
+* *Total risk:* An abstract value derived from the likelihood that the detected issue will negatively affect your infrastructure, and the impact on operations if that were to happen. Select one or more values from *Critical*, *Important*, *Moderate*, and *Low*.
```
```suggestion
* *Total risk:* An abstract value derived from the likelihood that the detected issue will negatively affect your infrastructure and the potential impact on operations. Select one or more values from *Critical*, *Important*, *Moderate*, and *Low*.
```
- remove unneeded comma before subordinating conjunction
- edit to reduce extra words.

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+* *Total risk:* An abstract value derived from the likelihood that the detected issue will negatively affect your infrastructure, and the impact on operations if that were to happen. Select one or more values from *Critical*, *Important*, *Moderate*, and *Low*.
+* *Impact:* The potential impact to the continuity of cluster operations. Select from *Critical*, *High*, *Medium*, and *Low*.
+* *Likelihood:* Filter by the potential for a negative impact to clusters if the recommendation comes to fruition. *Critical*, *High*, *Medium*, or *Low*.
+* *Category:* Select one or more categories of recommendations to concentrate on: *Service Availability*, *Performance*, *Fault Tolerance*, *Security*, *Best Practice*.
```
```suggestion
* *Category:* Filter by categories of recommendations. Select one or more values from *Service Availability*, *Performance*, *Fault Tolerance*, *Security*, and *Best Practice*.
```
Same comment as previously. In general, try to refer to these options in a consistent way for each item in the list.

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+* *Impact:* The potential impact to the continuity of cluster operations. Select from *Critical*, *High*, *Medium*, and *Low*.
+* *Likelihood:* Filter by the potential for a negative impact to clusters if the recommendation comes to fruition. *Critical*, *High*, *Medium*, or *Low*.
+* *Category:* Select one or more categories of recommendations to concentrate on: *Service Availability*, *Performance*, *Fault Tolerance*, *Security*, *Best Practice*.
+* *Status:* Set to only show *Enabled* recommendations by default, you can set the filter to show disabled recommendations.
```
```suggestion
* *Status:* Set to only show *Enabled* recommendations by default. You can also set the filter to show disabled recommendations.
```
- The meaning of this sentence is not clear to me. Why include the second part of the sentence? I don't know the functionality, so I'm not sure what I'm missing here. Is the setting for showing disabled recommendations something extra?

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+* *Likelihood:* Filter by the potential for a negative impact to clusters if the recommendation comes to fruition. *Critical*, *High*, *Medium*, or *Low*.
+* *Category:* Select one or more categories of recommendations to concentrate on: *Service Availability*, *Performance*, *Fault Tolerance*, *Security*, *Best Practice*.
+* *Status:* Set to only show *Enabled* recommendations by default, you can set the filter to show disabled recommendations.
+* *Clusters impacted:* Also set by default, you can filter to only recommendations currently impacting one or more clusters, and or show non-impacting recommendations.
```
```suggestion
* *Clusters impacted: Filter only those recommendations currently impacting one or more clusters. You can also show non-impacting recommendations. This setting is enabled by default.
```
- [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=punctuation-slashes#general-text), do not use the construction "and/or".

bburt-rh commented on modules/removing-filters-from-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,15 @@
+[id="removing-filters-from-insights-recommendations_{context}"]
+= Removing filters from Insights Advisor recommendations
+
+You can apply multiple filters to the list of recommendations and remove them individually or completely reset them.
```
```suggestion
You can apply multiple filters to the list of recommendations or remove them individually. You can also reset the filters.
```
- Edit for minimalism and to eliminate the awkward grammatical construction.

kblake2016 replied:
done

bburt-rh commented on modules/removing-filters-from-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,15 @@
+* Where the applied filters are displayed, remove applied filters individually, including the default filters, by clicking the *X* icon next to the filter.
+
+.Removing all non-default filters
+* You can remove only the filters that you applied, leaving the default filters in place, by clicking *Reset filters*.
```
```suggestion
* Click *Reset filters* to remove all the filters that you applied and to leave the default filters in place.
```
- Rewrite as a step in a procedure.

kblake2016 replied:
done

bburt-rh commented on modules/removing-filters-from-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,15 @@
+You can apply multiple filters to the list of recommendations and remove them individually or completely reset them.
+
+.Removing filters individually
+* Where the applied filters are displayed, remove applied filters individually, including the default filters, by clicking the *X* icon next to the filter.
```
```suggestion
* Where the applied filters are displayed, remove applied filters individually, including the default filters, by clicking the *X* icon next to the filter.
```
- Rewrite as a step in a procedure.
- Reword to use [the ISG guidance on referring to UI elements](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui). Not sure if this is a drop-down list but the description needs to use the standard UI element wording for whatever it is.

kblake2016 replied:
It's not a drop-down list. Each selected filter looks like an ovaltangular chip with a tiny x in the upper-left corner of it. All of the selected filters are visible at once. To remove individual filters, you click the x on the filter "chip." To remove them all (except for the defaults), you click (the blue words that aren't a button) *Reset filters*.

I rewrote each of these as procedures, but still used bullets since they're single steps.

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+
+* *Name:* Search for a recommendation by name.
+* *Total risk:* An abstract value derived from the likelihood that the detected issue will negatively affect your infrastructure, and the impact on operations if that were to happen. Select one or more values from *Critical*, *Important*, *Moderate*, and *Low*.
+* *Impact:* The potential impact to the continuity of cluster operations. Select from *Critical*, *High*, *Medium*, and *Low*.
```
```suggestion
* *Impact:* The potential impact to the continuity of cluster operations. Select one or more values from *Critical*, *High*, *Medium*, and *Low*.
```
- Not sure if this change is accurate, but the previous list item used this construction.

kblake2016 replied:
Good change. The meaning isn't changed. Done.

bburt-rh commented on modules/insights-advisor-recommendations-filters.adoc:
```diff
@@ -0,0 +1,24 @@
+* *Name:* Search for a recommendation by name.
+* *Total risk:* An abstract value derived from the likelihood that the detected issue will negatively affect your infrastructure, and the impact on operations if that were to happen. Select one or more values from *Critical*, *Important*, *Moderate*, and *Low*.
+* *Impact:* The potential impact to the continuity of cluster operations. Select from *Critical*, *High*, *Medium*, and *Low*.
+* *Likelihood:* Filter by the potential for a negative impact to clusters if the recommendation comes to fruition. *Critical*, *High*, *Medium*, or *Low*.
```
```suggestion
* *Likelihood:* Filter by the potential for a negative impact to clusters if the recommendation comes to fruition. Select one or more values from *Critical*, *High*, *Medium*, or *Low*.
```
- Same comment as for the previous item. In general, try to refer to these options in a consistent way for each item in the list.

kblake2016 replied:
Done

kblake2016 commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+[id="filtering-unnecessary-advisor-recommendations_{context}"]
+= Filtering Insights advisor recommendations
+
+As an OpenShift cluster manager, you can filter the recommendations that are displayed on the recommendations list. By applying filters, you can reduce the number of reported recommendations and concentrate on your highest priority recommendations.
```
Done

kblake2016 commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+You are logged in to the https://console.redhat.com/openshift[OpenShift Cluster Manager Hybrid Cloud Console].
+
+.Procedure
+. Navigate to the link:https://console.redhat.com/openshift/insights/advisor/recommendations[*Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*] page.
```
Done

kblake2016 commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+
+.Procedure
+. Navigate to the link:https://console.redhat.com/openshift/insights/advisor/recommendations[*Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*] page.
+. Using the filter input fields above the list of recommendations, select the *Category* filter type.
```
Done

bburt-rh commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+
+.Procedure
+. Navigate to *Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*.
+. In the left filter drop-down list, select the *Category* filter type.
+. Expand the right drop-down list and select the checkbox next to each category of recommendation you want to view. Leave the checkboxes for unnecessary categories clear.
```
I'm still wondering if there's some way to avoid referring to "left" and "right". [The IBM guidelines say not to refer to locations using sensory elements](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#location-of-interface-elements). Do these lists have a label or a heading of some kind that you can refer to?

kblake2016 replied:
I've changed the first instance from "left filter drop-down list" to "main, filter-type drop-down list," and the second instance from "right drop-down list" to "filter-value drop-down list." It still sounds inelegant to me, but I'll take this to my team and ask how to make this better with respect to the ISG and your feedback.

bburt-rh commented on modules/filtering-insights-advisor-recommendations.adoc:
```diff
@@ -0,0 +1,26 @@
+You are logged in to the https://console.redhat.com/openshift[OpenShift Cluster Manager Hybrid Cloud Console].
+
+.Procedure
+. Navigate to *Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*.
```
```suggestion
. Go to *Red Hat Hybrid Cloud Console* -> *OpenShift* -> *Advisor recommendations*.
```
Current [IBM Style word usage guidelines](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n) say to use _navigate_ "only when a more direct verb, such as _click_, _select_, _browse_, or _go to_, is not applicable".

kblake2016 replied:
Done

bburt-rh reviewed (approved):
Looks great now, @kblake2016! Very clean. Thanks so much for getting in all those changes. I've added a couple of other small things for you to consider changing before sending it on for merge review, but I'm going to go ahead and give this the `peer-review-done` label.

kblake2016 commented:
@bburt-rh , I've implemented your latest recommendations locally and updated the PR here. I'm preemptively requesting the merge review; I think that's the next step. Thanks again for your time and attention.

/label merge-review-needed

EricPonvelle commented:
/cherrypick enterprise-4.11

EricPonvelle commented:
/cherrypick enterprise-4.12

EricPonvelle commented:
/cherrypick enterprise-4.13

EricPonvelle commented:
/cherrypick enterprise-4.14

EricPonvelle commented:
/cherrypick enterprise-4.15
